Framework/Core/include/Framework/ASoA.h

```diff
@@ -125,9 +125,16 @@ class ColumnIterator : ChunkingPolicy
       mFirstIndex{0},
       mCurrentChunk{0}
   {
-    auto array = std::static_pointer_cast<arrow_array_for_t<T>>(mColumn->chunk(mCurrentChunk));
-    mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset();
-    mLast = mCurrent + array->length();
+    auto chunkToUse = getCurrentChunk();
+    if constexpr (std::is_same_v<arrow_array_for_t<T>, arrow::FixedSizeListArray>) {
+      auto array = std::static_pointer_cast<arrow_array_for_t<element_for_t<T>>>(chunkToUse);
+      mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset();
+      mLast = mCurrent + array->length();
```

Comment:
These two lines and 135-136 are the same. How about moving them out of the if?

Reply:
The two lines you highlight need to access array. Currently array is only defined within the if-clause, which is the reason why the two lines can not be moved out of the if.
On the other hand I need the if-clause to define the type of array, which is the reason why the definition of array can not be moved out of the if.
I can't think of a construction which would allow to avoid this.

Reply:
Found a possible solution - if you agree with the modifications in ArrowTypes.h.

Reply:
Can't you simply make getCurrentChunk templated and use `arrow_array_for_t<T>` to specify the trait? You can still use `if constexpr` inside getCurrentChunk, but at least the rest of the code is T-invariant.

Reply:
What about this solution - have auto do the job.

Reply:
... see latest changes (more simplifications) below.

Reply:
yes, that's what I meant, actually. Thanks. I have a few more cosmetic changes to ask (which I will post in a bit) and then we should be ready to go.